Sources/AccioKit/Services/XcodeProjectIntegrationService.swift

```diff
@@ -16,6 +16,28 @@ final class XcodeProjectIntegrationService {
         self.workingDirectory = workingDirectory
     }
 
+    func removeUnnecessaryGroups(keepingTargets targetsToKeep: [AppTarget]) throws {
+        guard let projectName = targetsToKeep.first?.projectName else { return }
+
+        let xcodeProjectPath = "\(workingDirectory)/\(projectName).xcodeproj"
+        let projectFile = try XcodeProj(path: Path(xcodeProjectPath))
+        let rootGroup = try projectFile.pbxproj.rootGroup()!
+        let dependenciesGroup = try rootGroup.group(named: Constants.xcodeDependenciesGroup) ?? rootGroup.addGroup(named: Constants.xcodeDependenciesGroup, options: .withoutFolder)[0]
+
+        let targetNames = targetsToKeep.map { $0.targetName }
+        let groupsToRemove = dependenciesGroup.children.filter { !targetNames.contains($0.name ?? "") }
+
+        guard !groupsToRemove.isEmpty else { return }
+
+        print("Removing unnecessary groups \(groupsToRemove.compactMap { $0.name }) from group 'Dependencies' ...", level: .info)
+
+        for groupToRemove in groupsToRemove {
+            dependenciesGroup.children.removeAll { $0 == groupToRemove }
```

Comment:
Here, you should load the franeworksbuildPhase of the target named like the group (if existing) and remove all frameworks within the group from the frameworks build phase.